packages/postgres-database/src/simcore_postgres_database/utils_folders.py

```diff
@@ -931,6 +933,51 @@ async def folder_remove_project(
         )
 
 
+def _get_subquery_my_access_rights(
+    access_via_gid: _GroupID, access_via_folder_id: _FolderID | None
+) -> ScalarSelect:
+    return (
+        sa.select(
+            sa.func.jsonb_build_object(
+                "read",
+                folders_access_rights.c.read,
+                "write",
+                folders_access_rights.c.write,
+                "delete",
+                folders_access_rights.c.delete,
+            ).label("my_access_rights"),
+        )
+        .where(
+            folders_access_rights.c.folder_id == access_via_folder_id
+            if access_via_gid and access_via_folder_id
+            else folders_access_rights.c.folder_id == folders.c.id
+        )
+        .where(folders_access_rights.c.gid == access_via_gid)
+        .correlate(folders)
+        .scalar_subquery()
+    )
+
+
+_SUBQUERY_ACCESS_RIGHTS: Final[ScalarSelect] = (
+    sa.select(
+        sa.func.jsonb_object_agg(
```

Comment:
oh... is this a json column? that is sad.

Reply:
No, there are 3 columns in the database. Here, he is creating a JSON field from those 3 columns, the way the frontend wants it.